lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp

```diff
@@ -785,13 +800,29 @@ void amf_ue_context_update_ue_sig_connection_state(
             AMF_APP_TIMER_INACTIVE_ID) {
       ue_context_p->m5_mobile_reachability_timer.sec =
           (amf_config.nas_config.t3512_min + 4 * 60);
-
-      if (ue_context_p->m5_mobile_reachability_timer.sec ==
-          AMF_APP_TIMER_INACTIVE_ID) {
-        ue_context_p->m5_mobile_reachability_timer.id = amf_app_start_timer(
-            ue_context_p->m5_mobile_reachability_timer.sec * 1000,
-            TIMER_REPEAT_ONCE, amf_app_handle_mobile_reachability_timer_expiry,
-            ue_context_p->amf_ue_ngap_id);
+      ue_context_p->m5_implicit_deregistration_timer.sec =
+          ue_context_p->m5_mobile_reachability_timer.sec;
+
+      // Start Mobile Reachability timer only if it is not running
+      if ((ue_context_p->m5_mobile_reachability_timer.id = amf_app_start_timer(
+               ue_context_p->m5_mobile_reachability_timer.sec * 1000,
+               TIMER_REPEAT_ONCE,
+               amf_app_handle_mobile_reachability_timer_expiry,
+               ue_context_p->amf_ue_ngap_id)) == -1) {
```

Comment:
Can we create and use a constant for -1 here?

Reply:
Addressed Review Comment